edgehog-device-runtime-forwarder/src/collection.rs

```diff
@@ -41,22 +43,70 @@ impl Connections {
         }
     }
 
-    /// Create a new [`Connection`] in case a new HTTP request is received.
-    #[instrument(skip(self, http_req))]
-    pub(crate) fn handle_http(
-        &mut self,
-        request_id: Id,
-        http_req: HttpRequest,
-    ) -> Result<(), Error> {
+    /// Handle the reception of an HTTP proto message from Edgehog.
+    #[instrument(skip_all)]
+    pub(crate) fn handle_http(&mut self, http: ProtoHttp) -> Result<(), Error> {
+        let ProtoHttp {
+            request_id,
+            http_msg,
+        } = http;
+
+        // the HTTP message can't be an http response
+        let http_req = http_msg.into_req().ok_or_else(|| {
+            error!("Http response should not be sent by Edgehog");
+            Error::WrongMessage(request_id.clone())
+        })?;
+
+        // before executing the HTTP request, check if it is an Upgrade request.
+        // if so, handle it properly.
+        if http_req.is_ws_upgrade() {
+            info!("Connection upgrade");
```

Comment:
Change to debug level and improve the log description.
```suggestion
            debug!("Connection upgrade ...");
```